onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -177,34 +177,37 @@ template <typename T, size_t N>
 constexpr size_t countof(T (&)[N]) { return N; }
 
 struct MyCustomKernel {
-  MyCustomKernel(OrtKernelInfo& /*info*/) {
+  MyCustomKernel(const OrtCustomOpApi& api, OrtKernelInfo& /*info*/) : api_(api) {
   }
 
   void GetOutputShape(OrtValue** inputs, size_t /*input_count*/, size_t /*output_index*/, OrtTensorTypeAndShapeInfo* info) {
-    OrtTensorDimensions dimensions(inputs[0]);
-    ORT_THROW_ON_ERROR(OrtSetDims(info, dimensions.data(), dimensions.size()));
+    OrtTensorDimensions dimensions(api_, inputs[0]);
+    ORT_THROW_ON_ERROR(api_.OrtSetDims(info, dimensions.data(), dimensions.size()));
   }
 
   void Compute(OrtValue** inputs, size_t /*input_count*/, OrtValue** outputs, size_t /*output_count*/) {
     const float* X;
     const float* Y;
-    ORT_THROW_ON_ERROR(OrtGetTensorMutableData(inputs[0], reinterpret_cast<void**>(const_cast<float**>(&X))));
-    ORT_THROW_ON_ERROR(OrtGetTensorMutableData(inputs[1], reinterpret_cast<void**>(const_cast<float**>(&Y))));
+    ORT_THROW_ON_ERROR(api_.OrtGetTensorMutableData(inputs[0], reinterpret_cast<void**>(const_cast<float**>(&X))));
```

Comment:
Well, as we talked before, I **hate** this. If I were the developer, if you ask me, do I want to put something in DLLMain, or rewrite the whole code base to change 'OrtXXX' to 'api.OrtXXX', I would definitely choose the first one.

Reply:
I made a small renaming change to help a bit.